compiler/passes/flattenFunctions.cpp

```diff
@@ -302,9 +305,13 @@ replaceVarUsesWithFormals(FnSymbol* fn, SymbolMap* vars) {
   if (fn->lifetimeConstraints)
     collectSymExprs(fn->lifetimeConstraints, symExprs);
 
-  form_Map(SymbolMapElem, e, *vars) {
-    if (Symbol* sym = e->key) {
-      ArgSymbol* arg  = toArgSymbol(e->value);
+  SymbolMapVector elts = sortedSymbolMapElts(*vars);
+  for (auto pair: elts) {
+    Symbol* key = pair.first;
+    Symbol* value = pair.second;
+
+    if (Symbol* sym = key) {
```

Comment:
I suggest style improvements in the above:
* Remove `key`, instead set `sym` directly to pair.first, perhaps with a comment saying "this is the key"
* Remove the conditional, as form_Map never produces null keys